app/models/user.rb

```diff
@@ -382,6 +388,20 @@ def is_new?
     Time.now - self.created_at <= NEW_USER_DAYS.days
   end
 
+  def is_heavy_self_promoter?
+    total_count = self.stories_submitted_count
+
+    if total_count < MIN_STORIES_CHECK_SELF_PROMOTION
+      false
+    else
+      authored = Story.where(
```

Comment:
Rails allows you to add to an association scope, so this could be `self.stories.where(:user_is_author => true).count`.